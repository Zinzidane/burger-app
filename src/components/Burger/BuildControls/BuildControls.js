import React from 'react';
import classes from './BuildControls.css';
import BuildControl from './BuildControl/BuildControl';

const controls = [
  { label: 'Salad', type: 'salad'},
  { label: 'Bacon', type: 'bacon'},
  { label: 'Meat', type: 'meat'},
  { label: 'Cheese', type: 'cheese'}
];

const buildControls = (props) => (
  <div className={classes.BuildControls}>
    {controls.map(ctrl => (
      <BuildControl
        added={() => props.ingredientAdded(ctrl.type)}
        key={ctrl.label}
        type={ctrl.type}
        label={ctrl.label} />
    ))}
  </div>
);

export default buildControls; 