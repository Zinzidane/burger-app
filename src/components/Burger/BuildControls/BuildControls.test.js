import React from 'react';
import BuildControls from './BuildControls';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<BuildControls />', () => {
    let wrapper;

    const props = {
      price: 0,
      purchasable: false,
      ordered: false,
      isAuth: false,
      disabled: {
        salad: false,
        bacon: false,
        cheese: false,
        meat: false
      }
    };
    beforeEach(() => {
      const controls = [
        { label: 'Salad', type: 'salad' },
        { label: 'Bacon', type: 'bacon' },
        { label: 'Cheese', type: 'cheese' },
        { label: 'Meat', type: 'meat' },
      ];
      wrapper = shallow(<BuildControls {...props}/>);
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});