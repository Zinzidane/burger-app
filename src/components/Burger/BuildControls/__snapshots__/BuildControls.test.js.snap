// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildControls /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <buildControls
    disabled={
        Object {
            "bacon": false,
            "cheese": false,
            "meat": false,
            "salad": false,
          }
    }
    isAuth={false}
    ordered={false}
    price={0}
    purchasable={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Current Price: 
          <strong>
                    0.00
          </strong>
</p>,
        Array [
          <buildControl
            added={[Function]}
            disabled={false}
            label="Salad"
            removed={[Function]}
/>,
          <buildControl
            added={[Function]}
            disabled={false}
            label="Bacon"
            removed={[Function]}
/>,
          <buildControl
            added={[Function]}
            disabled={false}
            label="Cheese"
            removed={[Function]}
/>,
          <buildControl
            added={[Function]}
            disabled={false}
            label="Meat"
            removed={[Function]}
/>,
        ],
        <button
          className={undefined}
          disabled={true}
          onClick={false}
>
          SIGN UP TO ORDER
</button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Current Price: ",
            <strong>
              0.00
</strong>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Current Price: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "0.00",
            },
            "ref": null,
            "rendered": "0.00",
            "type": "strong",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Salad",
        "nodeType": "function",
        "props": Object {
          "added": [Function],
          "disabled": false,
          "label": "Salad",
          "removed": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Bacon",
        "nodeType": "function",
        "props": Object {
          "added": [Function],
          "disabled": false,
          "label": "Bacon",
          "removed": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Cheese",
        "nodeType": "function",
        "props": Object {
          "added": [Function],
          "disabled": false,
          "label": "Cheese",
          "removed": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Meat",
        "nodeType": "function",
        "props": Object {
          "added": [Function],
          "disabled": false,
          "label": "Meat",
          "removed": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SIGN UP TO ORDER",
          "className": undefined,
          "disabled": true,
          "onClick": false,
        },
        "ref": null,
        "rendered": "SIGN UP TO ORDER",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Current Price: 
            <strong>
                        0.00
            </strong>
</p>,
          Array [
            <buildControl
              added={[Function]}
              disabled={false}
              label="Salad"
              removed={[Function]}
/>,
            <buildControl
              added={[Function]}
              disabled={false}
              label="Bacon"
              removed={[Function]}
/>,
            <buildControl
              added={[Function]}
              disabled={false}
              label="Cheese"
              removed={[Function]}
/>,
            <buildControl
              added={[Function]}
              disabled={false}
              label="Meat"
              removed={[Function]}
/>,
          ],
          <button
            className={undefined}
            disabled={true}
            onClick={false}
>
            SIGN UP TO ORDER
</button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Current Price: ",
              <strong>
                0.00
</strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Current Price: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0.00",
              },
              "ref": null,
              "rendered": "0.00",
              "type": "strong",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Salad",
          "nodeType": "function",
          "props": Object {
            "added": [Function],
            "disabled": false,
            "label": "Salad",
            "removed": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Bacon",
          "nodeType": "function",
          "props": Object {
            "added": [Function],
            "disabled": false,
            "label": "Bacon",
            "removed": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Cheese",
          "nodeType": "function",
          "props": Object {
            "added": [Function],
            "disabled": false,
            "label": "Cheese",
            "removed": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Meat",
          "nodeType": "function",
          "props": Object {
            "added": [Function],
            "disabled": false,
            "label": "Meat",
            "removed": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SIGN UP TO ORDER",
            "className": undefined,
            "disabled": true,
            "onClick": false,
          },
          "ref": null,
          "rendered": "SIGN UP TO ORDER",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
